streaming/src/java/org/apache/hive/streaming/HiveStreamingConnection.java

```diff
@@ -119,45 +105,35 @@
   private static final Logger LOG = LoggerFactory.getLogger(HiveStreamingConnection.class.getName());
 
   private static final String DEFAULT_METASTORE_URI = "thrift://localhost:9083";
-  private static final int DEFAULT_TRANSACTION_BATCH_SIZE = 1;
-  private static final int DEFAULT_HEARTBEAT_INTERVAL = 60 * 1000;
+  private static final int DEFAULT_TRANSACTION_BATCH_SIZE = 10;
```

Comment:
can this be done conditionally based on filesystem? for blob storage (eventually consistent) fs like s3a default to 1 and for hdfs (and other fs supporting immediate hflush) default to 10. This can be done in separate ticket (for easier tracking) as this change is unrelated to the ticket.

Reply:
👍 opened https://issues.apache.org/jira/browse/HIVE-20567